### PR TITLE
DOI schema gets exceptions for 5 hydrus identiiers

### PR DESCRIPTION
**Pull request by ndushay**
**NOTE:  Changes to openapi.yml require updating openapi.yml for sdr-api and dor-services-app and generating models - see README.**

## Why was this change made? 🤔

Adds DOI exceptions (from H2 work) to the DOI schema.

Part of #512

## How was this change tested? 🤨

rubular

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: openapi.yml [hunk 1]
```diff
@@ -896,8 +896,18 @@ components:
               # description: An alphabet or other notation used to represent a
               #   language or other symbolic system of the descriptive element value.
     DOI:
-      type: string
       description: Digital Object Identifier (https://www.doi.org)
+      oneOf:
+        - $ref: '#/components/schemas/DOI_pattern'
+        - $ref: '#/components/schemas/DOI_exceptions'
+    DOI_exceptions:
```
Comment (jcoyne) [inline openapi.yml:903]: I think this will have to be DoiExceptions, so that zeitwork can predict where this file is.

File: openapi.yml [hunk 2]
```diff
@@ -896,8 +896,18 @@ components:
...
+      type: string
+      description: pre-existing Digital Object Identifiers (https://www.doi.org) not matching the pattern
+      pattern: '^10\.(25740\/(VA90-CT15|syxa-m256|12qf-5243|65j8-6114)|25936\/629T-BX79)$'
```
Comment (justinlittman) [inline openapi.yml:906]: Would doing this as an `enum` instead of a `pattern` be clearer?

File: openapi.yml (hunk 1 above)
Comment (ndushay) [reply openapi.yml]: yep - got that error and fixed it.

File: openapi.yml (hunk 2 above)
Comment (ndushay) [reply openapi.yml:906]: good idea.  I will change.

Comment (ndushay) [reply openapi.yml]: answer:  DOI is NOT case sensitive, so pattern is better.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**